libensemble/gen_classes/sampling.py

```diff
@@ -31,11 +31,10 @@ class UniformSample(SampleBase):
     mode by adjusting the allocation function.
     """
 
-    def __init__(self, _, persis_info, gen_specs, libE_info=None):
-        self.persis_info = persis_info
-        self.gen_specs = gen_specs
-        self.libE_info = libE_info
+    def __init__(self, _=[], persis_info={}, gen_specs={}, libE_info=None, **kwargs):
+        super().__init__(_, persis_info, gen_specs, libE_info, **kwargs)
         self._get_user_params(self.gen_specs["user"])
+        self.gen_specs["out"] = [("x", float, (self.n,))]
```

Comment:
Interesting. This would overwrite any gen_specs['out'] supplied.

Reply:
Yeah, declaring input/output types is often implied by libE's interface to be something you can do; and something we should wholeheartedly support. but for UniformSample, for example, does it create anything _other than_ `x`s ?

I was thinking that it seems it doesn't matter if you decide to uniformly sample thetas instead:

```python
gen_specs["out"] = ["theta", float, (3,)]
```

the fields are often _nonetheless_ hardcoded like:

```python
H_o["x"] = ...
```

Maybe that's just because its a very simple gen. But this is also true for our other gens; you really can't customize ``gen_specs["out"]`` except for sizes.

So might as well hardcode the gen's fields, I guess, and make sure its clear to a user that its really the size thats customizable.

Though I guess if fields really were configurable, we'd have a gen that resembles:

```python
for field in gen_specs["out"]:
   if field == "theta":
      H["theta"] = do_experiment_with_theta()
   else:
      H["x"] = do_without_theta()
```